verifier/testreport.py

```diff
@@ -820,6 +823,9 @@ def check_simple_text_diffs(self, test_list, category):
             label = fail['label']
             actual = fail.get('result', None)
             expected = fail.get('expected', None)
+            if type(actual) != type(expected):
```

Comment:
I don't think this should be necessary because the actual & expected for the test cases we run should at least have the same type. If not, then our test case generation is severely wrong somehow.

However, we shouldn't gloss over that by ignoring the problem, so we should get rid of this `continue` here.

Reply:
Updating to note that results are not as expected due to a difference in data types. Then skip the string comparison characterizations.

Reply:
Restructuring to report correctly.